include/ppx/geometry.h

```diff
@@ -270,11 +284,15 @@ class Geometry
     Geometry::Buffer                                      mIndexBuffer;
     std::vector<Geometry::Buffer>                         mVertexBuffers;
     uint32_t                                              mPositionBufferIndex  = PPX_VALUE_IGNORED;
-    uint32_t                                              mNormaBufferIndex     = PPX_VALUE_IGNORED;
+    uint32_t                                              mNormalBufferIndex    = PPX_VALUE_IGNORED;
     uint32_t                                              mColorBufferIndex     = PPX_VALUE_IGNORED;
     uint32_t                                              mTexCoordBufferIndex  = PPX_VALUE_IGNORED;
     uint32_t                                              mTangentBufferIndex   = PPX_VALUE_IGNORED;
     uint32_t                                              mBitangentBufferIndex = PPX_VALUE_IGNORED;
+
+    bool               mInitialResizeMode  = false;
+    char*              mIndexBufferDataPtr = nullptr;
+    std::vector<char*> mVertexBuffersDataPtrs;
```

Comment:
I think it would be better to modify the behavior of `Geometry::Buffer` to do what you're wanting instead of adding new member vars that require additional tracking of the buffers separately.